src/redux/gatekeeperText.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+import type { ReduxMiddleware } from './store';
+
+function nameTemplate(fullText, playerName, bossName) {
+  return fullText.replace(/\[Name\]/g, playerName).replace(/\[BOSS\]/g, bossName);
+}
+
+const gatekeeperTextMiddleware: ReduxMiddleware = (store) => (next) => (action) => {
+  const result = next(action);
+  const currentState = store.getState();
+
+  if (
+    currentState.stage === 'fight' &&
+    ['perform_player_attack', 'perform_boss_attack'].includes(action.type)
+  ) {
+    if (currentState.attack) {
+      const attackObj = currentState.player.weapon.attacks.filter(
+        (attack) => currentState.attack && attack.name === currentState.attack.name,
+      );
```

Comment:
Do we need this? Like, can't we just read the text from `currentState.attack`? That object is the same type as the contents of the `player.weapon.attacks` and `boss.weapon.attacks` array. As a bonus, we'll get boss attacks displaying for free as well!